src/github/tests.rs

```diff
@@ -37,3 +35,32 @@ fn test_is_not_retryable_graphql_error() {
 
     assert!(!is_retryable_graphql_error(&error));
 }
+
+#[tokio::test]
+    async fn test_update_rate_limit_from_headers() {
```

Comment:
this only tests happy path, `rate_limit_guard` is not covered, same for jitter logic